src/init/ServerInitializer.ts

```diff
@@ -1,20 +1,31 @@
+import type { Server } from 'http';
+import { promisify } from 'util';
 import type { HttpServerFactory } from '../server/HttpServerFactory';
+import type { Finalizable } from './final/Finalizable';
 import { Initializer } from './Initializer';
 
 /**
  * Creates and starts an HTTP server.
  */
-export class ServerInitializer extends Initializer {
+export class ServerInitializer extends Initializer implements Finalizable {
   private readonly serverFactory: HttpServerFactory;
   private readonly port: number;
 
+  private server?: Server;
+
   public constructor(serverFactory: HttpServerFactory, port: number) {
     super();
     this.serverFactory = serverFactory;
     this.port = port;
   }
 
   public async handle(): Promise<void> {
-    this.serverFactory.startServer(this.port);
+    this.server = this.serverFactory.startServer(this.port);
+  }
+
+  public async finalize(): Promise<void> {
+    if (this.server) {
+      return promisify(this.server.close.bind(this.server))();
```

Comment:
Arrow function instead of `bind`?

Reply:
To get the typings correct that would be `promisify<void>((cb: any): any => this.server?.close(cb))()` (and that's already sort of cheating with the `any` typings) so personally I prefer the bind solution.

Reply:
Oh dear.